src/Dapr.PluggableComponents/Components/StateStore/ETagBulkDeleteRowMismatchException.cs

```diff
@@ -0,0 +1,50 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Globalization;
+using Grpc.Core;
+
+namespace Dapr.PluggableComponents.Components.StateStore;
+
+/// <summary>
+/// An exception that represents an occurrence of a bulk deletion not effecting the expected number of rows.
+/// </summary>
+/// <remarks>
+/// This exception should be thrown only from bulk delete operations.
+/// </remarks>
+public sealed class ETagBulkDeleteRowMismatchException : RpcException
```

Comment:
This one is not an etag error actually, you can rename to just `BulkDeleteRowMismatch`, it is thrown when a bulk delete is issue but the affected row count is different (no idea why)

Reply:
Good suggestion; I've renamed that exception and the underlying helper file (to be more generic).